yarn-project/p2p/src/services/peer-manager/peer_manager.ts

```diff
@@ -177,6 +179,28 @@ export class PeerManager implements PeerManagerInterface {
     }
   }
 
+  /**
+   * Processes scheduled disconnects during heartbeat.
+   *
+   * This batch processes all peers that have been marked for disconnect.
+   * preventing immediate disconnects that could cause libp2p state corruption.
+   */
+  private async processScheduledDisconnects() {
+    if (this.peersToBeDisconnected.size === 0) {
+      return;
+    }
+
+    const peersToDisconnect = Array.from(this.peersToBeDisconnected);
+    this.peersToBeDisconnected.clear();
+
+    this.logger.debug(`Processing ${peersToDisconnect.length} scheduled disconnects`);
+    try {
+      await Promise.all(peersToDisconnect.map(peerIdStr => this.disconnectPeer(peerIdFromString(peerIdStr))));
```

Comment:
```suggestion
      await Promise.all(peersToDisconnect.map(peerIdStr => this.disconnectPeer(peerIdFromString(peerIdStr))));
      this.logger.verbose(`Disconnected ${peersToDisconnect.length} peers`, { peersToDisconnect });
```
Just so we get more info about this in logs